services/circleci_service.py

```diff
@@ -0,0 +1,71 @@
+import requests
+
+
+class CircleciService:
+    def __init__(self, token, owner_id, github_org) -> None:
+        self.github_org = github_org
+        self.owner_id = owner_id
+        self.base_url = "https://circleci.com/api/v2/"
+        self.headers = {
+          "Circle-Token": token
+        }
+
+    def get_circleci_pipelines_for_repository(self, repo):
```

Comment:
I'd recommend moving the functions that are single requests into a client layer ♻️ Hides all the details of requests, headers and URLs from the service layer 🙈 and better highlights the business logic 💼